### DISCUSSION TITLE
okd 4.7 server down after etcd certificates expire

### DISCUSSION THREAD

**Discussion by michelsterckx** [okd-project/okd#1788] [Q&A]
Hi,
Our OKD 4.7 is down since a few days after the certificates used for the etcd database expired. These expired 3 years after the initial install. According to OKD [documentation](https://docs.okd.io/latest/security/certificate_types_descriptions/etcd-certificates.html) these should be automatically rotated which did not happen causing the cluster to go down.

etcd location: /etc/kubernetes/static-pod-resources/etcd-certs/secrets/etcd-all-peer/

I didn't find any documentation how to renew the certificates manually. Any idea what we could do to get this working again? 
regards,

**Comment by JaimeMagiera** [comment]
Please see this comment... 


https://github.com/okd-project/okd/discussions/1630#discussioncomment-6753842

**Reply by michelsterckx** [reply]
Thanks for the reply, Unfortunately it seems we are not in the same situation:

_[root@pvx9okdmastern1 ~]# export KUBECONFIG=/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig
[root@pvx9okdmastern1 ~]# oc get csr
The connection to the server localhost:6443 was refused - did you specify the right host or port?_

So the kube-apiserver is not running because the etcd cluster is down.  Any ideas?

**Comment by vrutkovs** [comment]
>According to OKD [documentation](https://docs.okd.io/latest/security/certificate_types_descriptions/etcd-certificates.html) these should be automatically rotated

Since OKD ~~4.9~~ 4.8 only

**Reply by michelsterckx** [reply]
If I look at the docs for 4.7 it does seem that's also 'system managed': [https://docs.okd.io/4.7/security/certificate_types_descriptions/etcd-certificates.html](url)

**Reply by vrutkovs** [reply]
They are system managed, as in "being created by system", but it doesn't mean they are being rotated. See [issues.redhat.com/browse/OCPBUGS-7198](https://issues.redhat.com/browse/OCPBUGS-7197).
